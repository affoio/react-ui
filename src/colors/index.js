const colors = {
  indigoBlue: {
    800: '#4718b4',
    700: '#623abf',
    600: '#8769ce',
    500: '#ac97dd',
    400: '#d1c5ec',
    200: '#ece7f7',
    300: '#e4ddf4',
    100: '#f6f3fb',
  },
  marigold: {
    800: '#ffca00',
    700: '#ffd226',
    600: '#ffdc59',
    500: '#ffe78c',
    400: '#fff2bf',
    300: '#fff7d9',
    200: '#fff9e5',
    100: '#fffcf2',
  },
  cerise: {
    800: '#e2086e',
    700: '#e62d84',
    600: '#ec5ea1',
    500: '#f290be',
    400: '#f8c1db',
    300: '#fbdaea',
    200: '#fce6f0',
    100: '#fdf2f8',
  },
  black: {
    800: '#000000',
    700: '#262626',
    600: '#595959',
    500: '#8c8c8c',
    400: '#bfbfbf',
    300: '#d9d9d9',
    200: '#e5e5e5',
    100: '#f2f2f2',
  },
  seaweed: {
    800: '#13c64f',
    700: '#36ce69',
    600: '#65da8c',
    500: '#95e5b0',
    400: '#c4f1d3',
    300: '#dcf7e5',
    200: '#e7f9ed',
    100: '#f3fcf6',
  },
  white: '#ffffff',
};

export default colors;
