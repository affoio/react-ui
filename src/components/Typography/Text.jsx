import React from 'react';
import { omit } from 'ramda';
import styled, { css } from 'styled-components';
import { oneOf, bool, string } from 'prop-types';

import componentFromProp from '../../helpers/hoc';
import { getTypographyFontStyle } from '../../theme/selectors';
import colors from '../../colors';

const getColorStyle = ({ inverse }) => `color: ${inverse ? colors.white : 'inherit'}`;

const Text = componentFromProp('tag');
Text.defaultProps = { tag: 'div' };

export const propTypes = {
  size: oneOf(['100', '200', '300', '400', '500', '600']),
  uppercase: bool,
  inverse: bool,
  fontWeight: string,
};

const TextStyled = styled(props => <Text {...omit(['size', 'uppercase', 'fontWeight', 'inverse'], props)} />)`
  display: inline;
  ${({ uppercase }) =>
    uppercase &&
    css`
      text-transform: uppercase;
    `}
  ${({ fontWeight }) =>
    css`
      font-weight: ${fontWeight};
    `}
  ${({ size, theme }) => getTypographyFontStyle(theme, size)}
  ${getColorStyle}
`;

TextStyled.defaultProps = {
  fontWeight: 'normal',
};

export default React.memo(TextStyled);
