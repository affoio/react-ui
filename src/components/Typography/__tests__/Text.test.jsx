import React from 'react';
import { render } from 'enzyme';
import renderer from 'react-test-renderer';
import Text from '../Text';
import colors from '../../../colors';

describe('<Text>', () => {
  it('renders correctly', () => {
    const tree = render(<Text>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Text>);
    expect(tree).toMatchSnapshot();
  });

  it('render inversed Text', () => {
    const tree = renderer
      .create(<Text inverse>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Text>)
      .toJSON();

    expect(tree).toHaveStyleRule(`color: ${colors.white}`);
  });

  it('render uppercased Text', () => {
    const tree = renderer
      .create(<Text uppercase>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Text>)
      .toJSON();

    expect(tree).toHaveStyleRule('text-transform: upppercase');
  });

  it('render Text with fontWeight', () => {
    const tree = renderer
      .create(
        <Text fontWeight="bold">Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Text>
      )
      .toJSON();
    expect(tree).toHaveStyleRule('font-weight: bold');
  });

  describe('renders with different sizes', () => {
    const sizes = ['100', '200', '300', '400', '500', '600'];

    sizes.forEach(size => {
      it(`render Text with size=${size}`, () => {
        const tree = render(
          <Text size={size}>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Text>
        );
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
