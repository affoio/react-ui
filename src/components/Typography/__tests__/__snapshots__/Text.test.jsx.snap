// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders correctly 1`] = `
.c0 {
  display: inline;
  font-weight: normal;
  font-size: 12px;
  line-height: 20px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Text> renders with different sizes render Text with size=100 1`] = `
.c0 {
  display: inline;
  font-weight: normal;
  font-size: 12px;
  line-height: 20px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Text> renders with different sizes render Text with size=200 1`] = `
.c0 {
  display: inline;
  font-weight: normal;
  font-size: 14px;
  line-height: 23px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Text> renders with different sizes render Text with size=300 1`] = `
.c0 {
  display: inline;
  font-weight: normal;
  font-size: 16px;
  line-height: 25px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Text> renders with different sizes render Text with size=400 1`] = `
.c0 {
  display: inline;
  font-weight: normal;
  font-size: 18px;
  line-height: 28px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Text> renders with different sizes render Text with size=500 1`] = `
.c0 {
  display: inline;
  font-weight: normal;
  font-size: 24px;
  line-height: 32px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Text> renders with different sizes render Text with size=600 1`] = `
.c0 {
  display: inline;
  font-weight: normal;
  font-size: 32px;
  line-height: 44px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;
